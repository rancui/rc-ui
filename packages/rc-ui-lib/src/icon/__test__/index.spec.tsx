import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { Icon } from '..';

const IconFont = Icon.createFromIconfontCN('//at.alicdn.com/t/font_2763890_w471tfudy4d.js');

describe('Icon', () => {
  afterEach(() => {
    cleanup();
    jest.restoreAllMocks();
  });

  it('should render icon with builtin icon name correctly', () => {
    const { container } = render(<Icon name="success" />);
    expect(container).toMatchSnapshot();
  });

  it('should render icon with url name correctly', () => {
    const { container } = render(<Icon name="https://img.yzcdn.com/icon.jpg" />);
    expect(container).toMatchSnapshot();
  });

  it('should render default slot correctly', () => {
    const { container } = render(<Icon name="success">Default Slot</Icon>);
    expect(container).toMatchSnapshot();
  });

  it('should change root tag after using tag prop', () => {
    const { container } = render(
      <Icon name="success" tag="div">
        Default Slot
      </Icon>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render dot correctly', () => {
    const { container } = render(<Icon name="success" dot />);
    expect(container.querySelector('.rc-badge')).toMatchSnapshot();
  });

  it('should render badge correctly', () => {
    const { container } = render(<Icon name="success" badge={{ content: 9 }} />);
    expect(container.querySelector('.rc-badge')).toMatchSnapshot();
  });

  it('should change icon size when using size prop', () => {
    const { container } = render(<Icon name="success" size={20} />);
    expect(getComputedStyle(container.querySelector('.van-icon')).fontSize).toEqual('20px');
  });

  it('should render correctly when createFromIconfontCN', () => {
    const { container } = render(<IconFont name="cuIcon-classify" color="#f44336" />);
    expect(container).toMatchSnapshot();
  });
});
