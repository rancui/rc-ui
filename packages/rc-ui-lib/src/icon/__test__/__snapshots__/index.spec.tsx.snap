// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should change root tag after using tag prop 1`] = `
<div>
  <div
    class="rc-badge__wrapper van-icon van-icon-success"
  >
    Default Slot
  </div>
</div>
`;

exports[`Icon should render badge correctly 1`] = `
<div
  class="rc-badge rc-badge--top-right rc-badge--fixed"
>
  9
</div>
`;

exports[`Icon should render correctly when createFromIconfontCN 1`] = `
<div>
  <i
    class="rc-badge__wrapper van-icon van-icon-cuIcon-classify"
    style="color: rgb(244, 67, 54);"
  >
    <svg
      fill="currentColor"
      height="1em"
      width="1em"
    >
      <use
        xlink:href="#cuIcon-classify"
      />
    </svg>
  </i>
</div>
`;

exports[`Icon should render default slot correctly 1`] = `
<div>
  <i
    class="rc-badge__wrapper van-icon van-icon-success"
  >
    Default Slot
  </i>
</div>
`;

exports[`Icon should render dot correctly 1`] = `
<div
  class="rc-badge rc-badge--top-right rc-badge--dot rc-badge--fixed"
/>
`;

exports[`Icon should render icon with builtin icon name correctly 1`] = `
<div>
  <i
    class="rc-badge__wrapper van-icon van-icon-success"
  />
</div>
`;

exports[`Icon should render icon with url name correctly 1`] = `
<div>
  <i
    class="rc-badge__wrapper van-icon"
  >
    <img
      alt="https://img.yzcdn.com/icon.jpg"
      class="rc-icon__image"
      src="https://img.yzcdn.com/icon.jpg"
    />
  </i>
</div>
`;
